Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -255,87 +262,53 @@ void UPSWorldSubsystem::AddProgressionStarActors()
 void UPSWorldSubsystem::OnTakeActorsFromPoolCompleted(const TArray<FPoolObjectData>& CreatedObjects)
 {
 	const FPSRowData& CurrentRowData = GetCurrentRow();
+	float AmountUnlocked = CurrentRowData.CurrentLevelProgression;;
 
-	float CurrentAmountOfUnlocked = 0;
-	float CurrentAmountOfLocked = 0;
-
-	//set updated amount of stars
-	if (CurrentRowData.CurrentLevelProgression >= CurrentRowData.PointsToUnlock)
-	{
-		// set required points (stars)  to achieve for a level  
-		CurrentAmountOfUnlocked = CurrentRowData.PointsToUnlock;
-		CurrentAmountOfLocked = 0;
-	}
-	else
-	{
-		// Calculate the unlocked against locked points (stars) 
-		CurrentAmountOfUnlocked = CurrentRowData.CurrentLevelProgression;
-		CurrentAmountOfLocked = CurrentRowData.PointsToUnlock - CurrentRowData.CurrentLevelProgression;
-	}
-
-	float integerPart;
-	float fractionalPart;
-	// Setup spawned widget
 	for (const FPoolObjectData& CreatedObject : CreatedObjects)
 	{
-		if (CurrentAmountOfUnlocked > 0)
+		AActor& SpawnedActor = CreatedObject.GetChecked<AActor>();
+		UStaticMeshComponent* MeshComponent = SpawnedActor.FindComponentByClass<UStaticMeshComponent>();
```

Comment:
`FindComponentByClass` is fine, but used only when there is no way to obtain component directly, like when it's not exposed in the engine files or dependency can not be included (when different modules etc).

In your case, you can simply create StarMeshComponent in your Star Actor and obtain it directly

please cast above directly to PSStarActor and get component directly
```
APSStarActor& SpawnedActor = CreatedObject.GetChecked<APSStarActor>();
UStaticMeshComponent* MeshComponent = SpawnedActor.GetStarMeshComponent();
```

Reply:
done as part of https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/3df84af0996411a050953b5ea3eeb9a58ed079c6
![image](https://github.com/user-attachments/assets/c125913b-c466-490a-b09e-b340a595f930)

Reply:
@h4rdmol 

The whole idea of this change was to replace `FindComponentByClass` usage by your StarMeshComponent property, but you still don't use it code doing search

Also,
- in actor blueprint, remove spaces from names, call it like 'StarMeshComponent', as unreal might have problems because of this
- in constructor, after calling CreateDefaultSubobject, on next line call `RootComponent = StarMeshComponent`, as there are known issues if not attach\set root
- in .h, format properly your StarMeshComponent property: add comment, add extra before next property, add `BlueprintReadWrite` specifer to be accessible on blueprint graphs

Reply:
achieved as part of: https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/ca8883f2e19e719a8226d94f24048c972fce855d
![image](https://github.com/user-attachments/assets/87e77883-0f09-4130-8984-213dcaeeb860)